src/pages/permissions/CreateIdentityModal.tsx

```diff
@@ -0,0 +1,119 @@
+import { FC, useState } from "react";
+import {
+  ActionButton,
+  Button,
+  Icon,
+  Input,
+  Modal,
+  Notification,
+} from "@canonical/react-components";
+import { useFormik } from "formik";
+import { createFineGrainedTlsIdentity } from "api/auth-identities";
+import { base64EncodeTokenDetails } from "util/helpers";
+import { useToastNotification } from "context/toastNotificationProvider";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+
+interface Props {
+  onClose: () => void;
+}
+
+const CreateIdentityModal: FC<Props> = ({ onClose }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [token, setToken] = useState<string | null>(null);
+  const [error, setError] = useState(false);
+  const toastNotify = useToastNotification();
+  const queryClient = useQueryClient();
+
+  const formik = useFormik({
+    initialValues: {
+      clientName: "",
+    },
+    onSubmit: (values) => {
+      setIsLoading(true);
+
+      createFineGrainedTlsIdentity(values.clientName)
+        .then((tokenPreResponse) => {
```

Comment:
```suggestion
        .then((response) => {
```
What is a token pre response?